packages/@o3r/core/schematics/service/index.ts

```diff
@@ -45,11 +44,16 @@ function ngGenerateServiceFn(options: NgGenerateServiceSchematicsSchema): Rule {
     const workspaceConfig = getWorkspaceConfig(tree);
     let packageName = destination;
     if (workspaceProject?.projectType !== 'application') {
-      let rec = '..';
-      while (!tree.exists(path.resolve(destination, rec, 'package.json')) && path.resolve(destination, rec) !== '/') {
-        rec = path.join('..', rec);
+      let currentDirectory = path.join(destination);
+      let parent = '..';
+      // Find the package.json file in the project directory
+      while (currentDirectory !== parent && !tree.exists(path.join(currentDirectory, 'package.json'))) {
```

Comment:
did you test that `currentDirectory !== parent` is correctly executing in the different OS?
It seems to be quite weak condition, maybe a `.include(parent)` would be a bit stronger?

Reply:
@kpanot : Changed it from === to includes